### PR TITLE
RBAC reference for Enterprise

### PR DESCRIPTION
**Pull request by jwitz**
Migrating from: https://github.com/astronomer/docs-archived/pull/473/files

### CODE REVIEW

File: enterprise/manage-platform-users.md [hunk 1]
```diff
@@ -183,3 +181,121 @@ To verify a user was successfully granted the SysAdmin role, ensure they can do
...
+- `deployment.variables.get`: View the Deployment's [environment variables](environment-variables.md)
+- `deployment.users.get`: View the list of users with access to the Deployment
+
+### Deployment Editor
+
+For a given Deployment, the Deployment Editor has all of the same default permissions as the Deployment Viewer, plus:
+
+- `deployment.airflow.user`: Airflow [user permissions](https://airflow.apache.org/docs/apache-airflow/stable/security/access-control.html#user) for all Deployments, including modifying task runs and DAG runs
+- `deployment.config.update`: Modify the Deployment's settings
+- `deployment.images.push`: Push code to the Deployment
```
Comment (paolaperaza) [inline enterprise/manage-platform-users.md:281]:
```suggestion
- `deployment.images.push`: Push code to the Deployment via the Astronomer CLI
```

File: enterprise/manage-platform-users.md [hunk 2]
```diff
@@ -183,3 +181,121 @@ To verify a user was successfully granted the SysAdmin role, ensure they can do
...
+
+- `deployment.airflow.get`: View the Airflow UI
+- `deployment.config.get`: View the Deployment's settings
+- `deployment.logs.get`: View the Deployment's logs
+- `deployment.images.pull`: Access the Deployment's running Docker image
+- `deployment.metrics.get`: View the Deployment's **Metrics** tab in the Astronomer UI
+- `deployment.serviceAccounts.get`: View any [service account](ci-cd.md#step-1-create-a-service-account) for the Deployment
+- `deployment.variables.get`: View the Deployment's [environment variables](environment-variables.md)
+- `deployment.users.get`: View the list of users with access to the Deployment
+
```
Comment (paolaperaza) [inline enterprise/manage-platform-users.md:274]:
```suggestion
- `deployment.variables.get`: View the Deployment's [environment variables](environment-variables.md)
- `deployment.users.get`: View the list of users with access to the Deployment

Note that a Deployment Viewer cannot cannot push code to a Deployment or modify Deployment configurations. For more information, reference Deployment Editor and Deployment Admin permissions below.
```

File: enterprise/manage-platform-users.md [hunk 3]
```diff
@@ -183,3 +181,121 @@ To verify a user was successfully granted the SysAdmin role, ensure they can do
...
+- `deployment.config.update`: Modify the Deployment's settings
+- `deployment.images.push`: Push code to the Deployment
+- `deployment.serviceAccounts.create`: Create a Deployment-level service account
+- `deployment.serviceAccounts.update`: Modify a Deployment-level service account
+- `deployment.serviceAccounts.delete`: Delete a Deployment-level service account
+- `deployment.variables.update`: Update the Deployment's [environment variables](environment-variables.md)
+
+### Deployment Admin
+
+For a given Deployment, the Deployment Editor has all of the same default permissions as the Deployment Viewer and the Deployment Admin, plus:
```
Comment (paolaperaza) [inline enterprise/manage-platform-users.md:289]:
```suggestion
For a given Deployment, the Deployment Admin has all of the same default permissions as the Deployment Viewer and the Deployment Editor, plus:
```

File: enterprise/manage-platform-users.md [hunk 4]
```diff
@@ -183,3 +181,121 @@ To verify a user was successfully granted the SysAdmin role, ensure they can do
...
+For a given Deployment, the Deployment Editor has all of the same default permissions as the Deployment Viewer, plus:
+
+- `deployment.airflow.user`: Airflow [user permissions](https://airflow.apache.org/docs/apache-airflow/stable/security/access-control.html#user) for all Deployments, including modifying task runs and DAG runs
+- `deployment.config.update`: Modify the Deployment's settings
+- `deployment.images.push`: Push code to the Deployment
+- `deployment.serviceAccounts.create`: Create a Deployment-level service account
+- `deployment.serviceAccounts.update`: Modify a Deployment-level service account
+- `deployment.serviceAccounts.delete`: Delete a Deployment-level service account
+- `deployment.variables.update`: Update the Deployment's [environment variables](environment-variables.md)
+
```
Comment (paolaperaza) [inline enterprise/manage-platform-users.md:286]:
```suggestion
- `deployment.variables.update`: Update the Deployment's [environment variables](environment-variables.md)

Note that a Deployment Editor cannot make changes to certain configurations in the Airflow UI. For more information, reference Deployment Admin permissions below.
```

File: enterprise/manage-platform-users.md [hunk 5]
```diff
@@ -183,3 +181,121 @@ To verify a user was successfully granted the SysAdmin role, ensure they can do
...
+- `workspace.config.update`: Modify the Workspace, including Workspace Name, Description, and user access
+- `workspace.deployments.create`: Create a Deployment in the Workspace
+- `workspace.serviceAccounts.create`: Create a Workspace-level service account
+- `workspace.serviceAccounts.update`: Modify a Workspace-level service account
+- `workspace.serviceAccounts.delete`: Delete a Workspace-level service account
+
+### Workspace Admin
+
+For a given Workspace, the Workspace Editor has all of the same default permissions as the Workspace Viewer and Workspace Editor, plus:
```
Comment (paolaperaza) [inline enterprise/manage-platform-users.md:256]:
```suggestion
For a given Workspace, the Workspace Admin has all of the same default permissions as the Workspace Viewer and Workspace Editor, plus:
```

**Review by paolaperaza — APPROVED** [review]
A few minor comments here but LGTM - will let you finish this out and merge @jwitz! Make sure to just port these over to prior versions.

**Comment by jwitz** [comment]
Sweet, gonna merge! Confirmed [via Slack](https://astronomer.slack.com/archives/C02NZELT2/p1641383147000500)  that there are some permissions not explicitly tied to the default.yaml file. Specifically, I couldn't find any one permission that enabled Workspace Admins to complete CRUD operations on Deployments in their workspace. But that is the confirmed behavior, so added a note about it. If we catch any more of these, we should add them here.

**Pull request merged**